tests/test_provider_configuration.py

```diff
@@ -57,41 +60,154 @@ def test_should_not_fetch_provider_metadata_if_given(self):
+            'redirect_uris': ['https://client.example.com/redirect', 'https://client.example.com/logout']})
 
+        post_logout_redirect_uris = ['https://client.example.com/logout']
         provider_config = ProviderConfiguration(
             provider_metadata=self.provider_metadata(registration_endpoint=registration_endpoint),
-            client_registration_info=ClientRegistrationInfo())
+            client_registration_info=ClientRegistrationInfo(
+                post_logout_redirect_uris=post_logout_redirect_uris))
 
-        extra_args = {'extra_args': 'should be passed'}
-        redirect_uris = ['https://client.example.com/redirect']
-        provider_config.register_client(redirect_uris, extra_args)
+        provider_config.register_client(Client(CLIENT_AUTHN_METHOD), 'https://client.example.com/redirect')
         assert provider_config._client_metadata['client_id'] == 'client1'
         assert provider_config._client_metadata['client_secret'] == 'secret1'
-        assert provider_config._client_metadata['redirect_uris'] == redirect_uris
-
-        expected_registration_request = {'redirect_uris': redirect_uris}
-        expected_registration_request.update(extra_args)
-        assert json.loads(responses.calls[0].request.body.decode('utf-8')) == expected_registration_request
+        assert provider_config._client_metadata['redirect_uris'] == [
+            'https://client.example.com/redirect', 'https://client.example.com/logout']
+        assert provider_config._client_metadata[
+                   'post_logout_redirect_uris'] == post_logout_redirect_uris
 
     def test_should_not_register_dynamic_client_if_client_metadata_is_given(self):
         client_metadata = ClientMetadata(client_id='client1',
                                          client_secret='secret1',
                                          redirect_uris=['https://client.example.com/redirect'])
         provider_config = ProviderConfiguration(provider_metadata=self.provider_metadata(),
                                                 client_metadata=client_metadata)
-        provider_config.register_client([])
+        provider_config.register_client(None, None)
         assert provider_config._client_metadata == client_metadata
 
     def test_should_raise_exception_for_non_registered_client_when_missing_registration_endpoint(self):
         provider_config = ProviderConfiguration(provider_metadata=self.provider_metadata(),
                                                 client_registration_info=ClientRegistrationInfo())
+        assert provider_config._provider_metadata['registration_endpoint'] is None
         with pytest.raises(ValueError) as exc_info:
-            provider_config.register_client([])
-        assert 'registration_endpoint' in str(exc_info.value)
+            provider_config.register_client(None, None)
+
+    @responses.activate
+    def test_register_client_should_register_dynamic_client_if_initial_access_token_present(self):
```

Comment:
This test should also assert the given `registration_token` was actually passed in the registration request:
Something like the following (which I've not tested it, so not 100% this is the way to access the headers from the captured request):
```python
assert responses.calls[0].request.headers['Authorization'] == 'Bearer initial_access_token'
```

Reply:
```python
assert provider_config._client_metadata['registration_access_token'] == 'registration_access_token1'
```
This has been asserted here.

Reply:
Those are not the same 🙂 :
* `assert responses.calls[0].request.headers['Authorization'] == 'Bearer initial_access_token'` <- checks the specified access token was included in the correct way in the registration request
* `assert provider_config._client_metadata['registration_access_token'] == 'registration_access_token1'` <- checks the parameter `registration_access_token` was returned in the registration response (also this parameter is not required to be returned by the IdP).